### ISSUE TITLE
Can't change meeting address

### ISSUE THREAD

**Issue by rstelma** [code4recovery/12-step-meeting-list#127]
**Describe the bug**
I'm having trouble changing the address. I put in the new, correct address and I click on update in Word Press and the address just reverts back to the old address. There are quite a few meetings at this location. There is a checkbox coming up that says if I want to change the address for all meetings at this location but it doesn't matter if I check it or not. The address keeps reverting back to this address as soon as I click out of the address field.

http://district12.org/meetings/breakfast-bunch-2/?tsml-day=5&tsml-region=hammond

**To Reproduce**
1. Go to Edit the meeting
2. Click in address location. Change the address. 
3. Update (click on)
4. Old address is saved and I cannot save the new address

**Expected behavior**
I expect to be able to change the address

**Screenshots**
![oldaddress](https://user-images.githubusercontent.com/68346976/87578294-d4c1fb80-c699-11ea-96ba-07163cf0c25d.JPG)

![oldaddressonline](https://user-images.githubusercontent.com/68346976/87578380-f3c08d80-c699-11ea-8bcf-49fe8edc715d.JPG)

![Status](https://user-images.githubusercontent.com/68346976/87578719-721d2f80-c69a-11ea-861a-f6a7cc038efb.JPG)

**Desktop (please complete the following information):**
 - OS: Windows 7
 - Browser Chrome Version 
 - Version 83.0.4103.116 (Official Build) (64-bit) 
 - Browser: Firefox 
 - Version 78.0.2 (64-bit)

**Smartphone (please complete the following information):**
N/A

**Additional context**

**Comment by brianw-area24** [comment]
I tried this on a test site, and I'm not able to duplicate results you describe. I can think of a couple possible causes, however, need a little more information before I can troubleshoot any farther. Looks like the old address is 411 W Charles St, Hammond, LA 70401, USA.

What is the new address you're trying to put in?
Are you wanting to change the address for all the meetings at this address, or just one of them?

**Comment by rstelma** [comment]
Hi Brian,

Thank you.

The new address is: 404 W Thomas St, Hammond, LA 70401

I need to change the address for all meetings but I would be happy to be
able to change even one of them. I've tried saving for both all of the
meetings and for only one.

Can you tell me where in the database the addresses are located? I tried
finding the addresses but no luck. I'm very experienced with SQL and I have
backups of the database just in case.

Thank you,
Richard

**Comment by brianw-area24** [comment]
Thanks for that address. I went to Google maps, and entered that address, and Google seemed to like the new address, which was the first thing I wanted to test.

As you are aware, interacting with the database directly can be dangerous, and you can easily break a site if you're not careful. The address and the long/lat data is stored in the wp_postmeta table. First thing to do is find the post ID for your location by querying wp_posts for post_type = 'tsml_location'. Once you have the correct ID, query wp_postmeta for post_id = <your id number>

My fear is that if you change the address directly in the database, when you modify one of the meetings in the future, it will switch back to the existing address. So there are a couple more things we can try.

1. Try creating a new meeting, using the new address on Thomas St. My guess is that it will switch the address to the existing Charles Street address. Then try creating a meeting at a completely different address, and my guess is that it will take this different address. You should then delete these tests message.
2. There is some code you can add to your theme's functions.php file for the new address. See https://wordpress.org/plugins/12-step-meeting-list/#faq, look at the 3rd question - "I need to correct a meeting address or change a pin's location"

I'm hoping that this second option will help you.

**Comment by rstelma** [comment]
I know of the dangers of fooling in the database if you don't know what
you're doing. I've working with databases for 18 years, designing,
building, restoring, moving, configuring, etc. I don't consider myself
beyond making a mistake but I'm extremely careful. I would rather get the
app fixed. I would also like to know where that checkbox value is stored,
the one that says "Apply this updated address to all meetings at this
location".

1. I created a meeting on Thomas St. I clicked on Publish. Saved fine and I
was able to Preview the meeting with the correct address, However, on the
edit page it shows the address as the 411 W Charleston address. If I don't
Update it will keep the 404 W Thomas address. When I Update of course it
changes the address to the 411 because that's what's in the text box. The
meeting I created at a completely different address is able to be edited. I
put in 29405 Teel Rd, published and was able to change it 39440 Teel Rd.
Seems like if I enter 404 W Thomas and Publish it will forever be 411 W
Charles St.
2. I added

if (function_exists('tsml_custom_types')) {
    tsml_custom_addresses(array(
        '5 Avenue Anatole France, 75007 Paris, France' => array(
            'formatted_address' => '5 Avenue Anatole France, 75007
Paris, France',
            'city' => 'Paris',
            'latitude' => 48.858372,
            'longitude' => 2.294481,
        ),
    ));
}

to functions.php and still having the same problem. No change. Unable to
change an address if has ever been assigned the address 404 W Thomas St. I
have a feeling I might have gotten here by using that checkbox and trying
to change the address at all locations at once. There's some code somewhere
that's preventing this from saving. There is a bug. I want to write a query
so that I can see how those locations are linked and see what the value is
the bit value of that linking bit. If I could show you those values you
might be able to recreate the problem.

Very much appreciated.

Richard
